### PR TITLE
Tried to replace jobintentservice with workmanager

### PR DESCRIPTION
**Pull request by arjundevasi**
Please review changes and point mistakes,

### CODE REVIEW

File: android/src/app/organicmaps/MwmJobIntentService.java [hunk 1]
```diff
@@ -8,32 +9,41 @@
...
-    if (!app.arePlatformAndCoreInitialized())
-    {
-      Logger.w(TAG, "Application is not initialized, ignoring " + intent);
-      return;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+import java.util.logging.Logger;
+
+public abstract class MwmJobIntentService extends Worker {
```
[the remark this answers is not in the document]
Comment (arjundevasi) [reply android/src/app/organicmaps/MwmJobIntentService.java:17]: at doWork()

File: android/src/app/organicmaps/MwmJobIntentService.java [hunk 2]
```diff
@@ -8,32 +9,41 @@
...
+            MwmApplication app = MwmApplication.from(this);
+            String msg = "onHandleWork: " + intent;
+            Logger.i(TAG, msg);
+            CrashlyticsUtils.INSTANCE.log(Log.INFO, getTag(), msg);
+
+            onHandleWorkInitialized(intent);
+
+            return Result.success();
+        } catch (Throwable throwable) {
+            Logger.w(TAG, "Application is not initialized, ignoring " + intent);
```
Comment (biodranik) [inline android/src/app/organicmaps/MwmJobIntentService.java:43]: When and why is this situation possible?

File: android/src/app/organicmaps/MwmJobIntentService.java [hunk 3]
```diff
@@ -8,32 +9,41 @@
...
-    onHandleWorkInitialized(intent);
-  }
+    protected abstract void onHandleWorkInitialized(@NonNull Intent intent);
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        Context applicationContext = getApplicationContext();
+
+        try {
```
Comment (biodranik) [inline android/src/app/organicmaps/MwmJobIntentService.java:32]: We use different code style formatting. Please check docs/JAVA_STYLE.md

File: android/src/app/organicmaps/MwmJobIntentService.java [hunk 4]
```diff
@@ -8,32 +9,41 @@
...
     }
 
-    onHandleWorkInitialized(intent);
-  }
+    protected abstract void onHandleWorkInitialized(@NonNull Intent intent);
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        Context applicationContext = getApplicationContext();
```
Comment (biodranik) [inline android/src/app/organicmaps/MwmJobIntentService.java:30]: All non-modified vars should have `final` keyword for better readability.

**Review by biodranik — COMMENTED** [review]
How did you test it and confirmed that it works?

File: android/src/app/organicmaps/MwmJobIntentService.java (hunk 4 above)
Comment (arjundevasi) [reply android/src/app/organicmaps/MwmJobIntentService.java:30]: I did not test it.
i'm sorry for wasting your time.

Comment (biodranik) [reply android/src/app/organicmaps/MwmJobIntentService.java:30]: Please test it by adding additional logs, and by trying to edit some features in OSM (but you need to build a beta build for that, not a debug one).

Comment (arjundevasi) [reply android/src/app/organicmaps/MwmJobIntentService.java:30]: ok, I'll do that.
Thanks for guidance

**Comment by arjundevasi** [comment]
Understood, please allow me some time.

**Comment by arjundevasi** [comment]
The background process in `osmuploadservice`  is `editor.uploadchanges` or something else?

**Comment by arjundevasi** [comment]
```
package app.organicmaps.background;

import android.content.Context;


import androidx.annotation.NonNull;

import androidx.work.Constraints;
import androidx.work.NetworkType;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkManager;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import app.organicmaps.editor.Editor;


public class OsmUploadService extends Worker
{
  public OsmUploadService(@NonNull Context context, @NonNull WorkerParameters workerParams) {
    super(context, workerParams);
  }

  private static final String TAG = OsmUploadService.class.getSimpleName();

  /**
   * Starts this service to upload map edits to osm servers.
   */
   
  public static void startActionUploadOsmChanges(@NonNull Context context)
  {

    WorkManager workManager = WorkManager.getInstance(context);

    Constraints constraints = new Constraints.Builder()
            .setRequiredNetworkType(NetworkType.UNMETERED)
            .build();

    OneTimeWorkRequest oneTimeWorkRequest = new OneTimeWorkRequest.Builder(OsmUploadService.class)
            .setConstraints(constraints)
                    .build();

    workManager.enqueue(oneTimeWorkRequest);

  }

  @NonNull
  @Override
  public Result doWork() {

    final Context applicationContext = getApplicationContext();

    try{
      Editor.uploadChanges(applicationContext);
      return Result.success();
    } catch (Throwable throwable){
      return Result.failure();
    }

  }
}

```

**Comment by arjundevasi** [comment]
Please respond

**Comment by biodranik** [comment]
See #4691

**Pull request closed without merging**